opcua/common/xmlexporter.py

```diff
@@ -228,7 +229,26 @@ def add_variable_common(self, node, el):
         if dim.Value.Value:
             el.attrib["ArrayDimensions"] = ",".join([str(i) for i in dim.Value.Value])
         el.attrib["DataType"] = dtype_name
-        value_to_etree(el, dtype_name, dtype, node)
+        self.value_to_etree(el, dtype_name, dtype, node)
+
+    def _get_variable_basetype(self, datatype):
+        """
+        Looks up the base datatype of the provided datatype. 
+        The base datatype is either:
+        A primitive type (ns=0, i<=21) or a complex one (ns=0 i>21 and i<=30) like Enum and Struct.
+        
+        Args:
+            datatype: NodeId of a datype of a variable
+        Returns:
+            NodeId of datatype base or None in case base datype can not be determined
+        """
+        if datatype.NamespaceIndex == 0 and type(datatype) == ua.NumericNodeId and datatype.Identifier <= 30:
+            return datatype
+        dtype_supers_nodes = get_node_supertypes(self.server.get_node(datatype), includeitself=True, skipbase=True)
+        dtype_supers_nodeids = [node.nodeid for node in dtype_supers_nodes if node.nodeid.NamespaceIndex == 0 and  node.nodeid.Identifier <= 30]
+        if len(dtype_supers_nodeids) >= 1:
+            return dtype_supers_nodeids[0]
+        return None
```

Comment:
Did you copy the method from ua_utils.py? Can't you call it from ua_utils.py? you can modify/split the one in ua_utils.py if you need so

I think we should raise an exception if no data type is found, this is not normal. so raising an exception is correct and clearly shows there is an issue